p2panda-group/src/crypto/provider.rs

```diff
@@ -0,0 +1,275 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use std::sync::RwLock;
+
+use rand_chacha::rand_core::{SeedableRng, TryRngCore};
+use thiserror::Error;
+
+use crate::crypto::{aead, ed25519, hkdf, hpke, x25519, xchacha20, xeddsa};
+use crate::traits::{CryptoProvider, RandProvider, XCryptoProvider};
+
+#[derive(Debug)]
+pub struct Crypto {
```

Comment:
I'm not sure about the naming of the "default" struct implementing all the traits. Should it be `Crypto`? `Provider`? `DefaultProvider`? .. they are all quite too generic but maybe it's fine?